sources/src/TorrentPlugin.cpp

```diff
@@ -278,40 +242,31 @@ void TorrentPlugin::on_add_peer(const libtorrent::tcp::endpoint & endPoint, int
 
 void TorrentPlugin::pieceRead(const libtorrent::read_piece_alert * alert) {
 
-    // There should be at least one peer registered for this piece, unless they have disconnected
-    auto it = _outstandingLoadPieceForBuyerCalls.find(alert->piece);
+    // There should be a registeration for this piece, unless we have left selling mode
+    auto it = _outstandingLoadPieceForBuyers.find(alert->piece);
 
-    if(it == _outstandingLoadPieceForBuyerCalls.cend()) {
+    if(it == _outstandingLoadPieceForBuyers.cend()) {
 
         std::clog << "Ignoring piece read, must be for some other purpose." << std::endl;
         return;
     }
 
-    // Make a callback for each peer registered
-    const std::set<libtorrent::peer_id> & peers = it->second;
-
-    // Iterate peers
-    for(auto peerId : peers) {
-        auto peerPlugin = peer(peerId);
-        auto endPoint = peerPlugin->endPoint();
+    // Remove registeration
+    _outstandingLoadPieceForBuyers.erase(it);
 
-        // Make sure reading worked
-        if(alert->ec) {
+    // Make sure reading worked
+    if(alert->ec) {
 
-            std::clog << "Failed reading piece" << alert->piece << "for" << libtorrent::print_address(endPoint.address()).c_str() << std::endl;
-            assert(false);
+        std::clog << "Failed reading piece" << alert->piece << std::endl;
```

Comment:
I guess in principle this could actually fail, and we should handle it more gracefully`? Why not add an issue for this, no need to get into any analysis at this point though.